cpp/include/cudf/io/datasource.hpp

```diff
@@ -31,10 +31,15 @@
 #include <arrow/filesystem/s3fs.h>
 #pragma nv_diag_default 611
 
+// We disable warning 2810 to workaround the compile issue (warning treated as error):
+// result.h(263): error #2810-D: ignoring return value type with "nodiscard" attribute
+#pragma nv_diag_suppress 2810
+#include <arrow/result.h>
```

Comment:
Can we file an Arrow issue for this?

Reply:
The underlying cause is an nvcc issue, which I believe @vyasr has submitted an issue for.

Reply:
I think so. Also the issue that we did similarly in the past (warning 611: https://github.com/rapidsai/cudf/pull/12465/files/1ab9604856ff666d87847e569af8b2c4b49d5dbc#diff-e87e4906a9ed3d9d0c2c663d5e648ec99950d9042fb297a6a195a417276dc55fR29-R32).

Reply:
We had similar workaround in the past in https://github.com/rapidsai/cudf/pull/8916#issuecomment-1299410665, not sure if all these issues have been reported.

Reply:
Yes, I have an open nvbug for this issue.